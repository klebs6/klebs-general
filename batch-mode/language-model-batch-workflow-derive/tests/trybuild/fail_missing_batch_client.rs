// ---------------- [ File: tests/trybuild/fail_missing_batch_client.rs ]
//
// Because the macro requires `#[batch_client]` on exactly one field, failing to provide it
// should yield a compile error. The harness calls `t.compile_fail(...)`, expecting an error.

use language_model_batch_workflow_derive::LanguageModelBatchWorkflow;
use batch_mode_batch_workflow::*;

// Missing `#[batch_client]` => should fail at compile time with a clear error about
// “Missing required `#[batch_client]`.”

// "fail_missing_batch_client.rs"
#[derive(LanguageModelBatchWorkflow)]
struct MissingClient {
    #[batch_workspace]
    ws: std::sync::Arc<BatchWorkspace>, // <-- now meets the new type requirement
    #[model_type]
    mt: LanguageModelType,
}

// We define the custom error type, so that part is satisfied:
struct MyErr;

// Because there's no #[batch_client], the derive macro logic should fail parse.
fn main() {}
