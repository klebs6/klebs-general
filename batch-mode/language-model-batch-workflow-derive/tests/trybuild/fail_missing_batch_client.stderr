error: cannot find attribute `batch_workspace` in this scope
  --> tests/trybuild/fail_missing_batch_client.rs:14:7
   |
14 |     #[batch_workspace]
   |       ^^^^^^^^^^^^^^^

error: cannot find attribute `expected_content_type` in this scope
  --> tests/trybuild/fail_missing_batch_client.rs:16:7
   |
16 |     #[expected_content_type]
   |       ^^^^^^^^^^^^^^^^^^^^^

error: cannot find attribute `model_type` in this scope
  --> tests/trybuild/fail_missing_batch_client.rs:18:7
   |
18 |     #[model_type]
   |       ^^^^^^^^^^
   |
note: `model_type` is imported here, but it is a module, not an attribute
  --> tests/trybuild/fail_missing_batch_client.rs:7:5
   |
7  | use batch_mode_batch_workflow::*;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused import: `language_model_batch_workflow_derive::LanguageModelBatchWorkflow`
 --> tests/trybuild/fail_missing_batch_client.rs:6:5
  |
6 | use language_model_batch_workflow_derive::LanguageModelBatchWorkflow;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
